src/WebJobs.Extensions.NotificationHubs/Bindings/NotificationHubAsyncCollector.cs

```diff
@@ -4,23 +4,39 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.NotificationHubs;
+using Microsoft.Azure.WebJobs.Host;
 
 namespace Microsoft.Azure.WebJobs.Extensions.NotificationHubs
 {
     internal class NotificationHubAsyncCollector : IAsyncCollector<Notification>
     {
         private INotificationHubClientService _notificationHubclientService;
         private string _tagExpression;
+        private bool _enableTestSend;
+        private TraceWriter _traceWriter;
 
-        public NotificationHubAsyncCollector(INotificationHubClientService clientService, string tagExpression)
+        public NotificationHubAsyncCollector(INotificationHubClientService clientService, string tagExpression, bool enableTestSend, TraceWriter traceWriter)
         {
             _notificationHubclientService = clientService;
             _tagExpression = tagExpression;
+            _enableTestSend = enableTestSend;
+            _traceWriter = traceWriter;
         }
 
         public async Task AddAsync(Notification item, CancellationToken cancellationToken = default(CancellationToken))
         {
-            await _notificationHubclientService.SendNotificationAsync(item, _tagExpression);
+            NotificationOutcome notificationOutcome = await _notificationHubclientService.SendNotificationAsync(item, _tagExpression);
+            if (_enableTestSend)
+            {
+                _traceWriter.Info(string.Format("TrackingId:{0}", notificationOutcome.TrackingId));
```

Comment:
A few suggestions

1. Since these logs can all be interleaved between each other, you may want to build up a single string and call `.Info()` once so that it's all in one place. It can get confusing otherwise. You can use newlines to break it up to look nicer.
2. You may want to start that string with something that indicates it's coming from Notification Hubs. Otherwise you'll just see a log with `TrackingId:` and that may not have enough context.
3. Right now you're only logging for the case where you have results. You should also log if results are null or if results are empty. This looks like a debugging scenario that will only be turned on occasionally so you can be verbose about exactly what's happening.